lib/runtime/addon.js

```diff
@@ -341,4 +340,34 @@ export default class Addon {
     return configModule && (configModule.default || configModule);
   }
 
+  get blueprintNames() {
+    let names = fs.readdirSync(path.join(this.dir, 'blueprints'));
+    return names;
+  }
+
+  get blueprints() {
+    return this.blueprintNames.map((name) => {
+      let blueprintPath = path.join(this.dir, 'blueprints', name);
+      let Print = require(blueprintPath).default;
+      return new Print({ name, addon: this, description: Print.description });
+    });
+  }
+
+  hasBlueprint(name) {
+    const names = this.blueprintNames;
+    const blueprint = names.includes(name);
+
+    return Boolean(blueprint);
+  }
+
+  getBlueprint(name) {
+    try {
+      let blueprintPath = path.join(this.dir, 'blueprints', name);
+      let Blueprint = require(blueprintPath).default;
+      return new Blueprint({ name, addon: this });
```

Comment:
Same here - rather than instantiating a new Blueprint instance every time, we should be pulling the same object (either by eagerly instantiating all the Blueprints on initialization of the Addon instance and using that array, or by caching/memoizing these functions).